### PR TITLE
test.pypi.org deployment

### PR DESCRIPTION
**Pull request by weakcamel**
* add (encrypted) credentials to upload to test.pypi.org
* assign "dev" version for builds from branches and release
   version for Git tags; when using Makefile, always
   build a dummy dev version
* setup.py: format of the long description is now text/markdown
* add a link to API Reference in README

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,35 @@
...
+    distributions: sdist bdist_wheel
+    server: https://test.pypi.org/legacy/
+    "on":
+      all_branches: true
+      tags: false
+
+env:
+  global:
+    # TEST_PYPI_ORG_TOKEN
+    - secure: TUQef9pZbFtOBGLQkAT0pzRmYjTR6LJOucM2x+kMfGfueN1khiwx/LmcShgCha+7kH43v0/1q9Iyk6+xQIfe7TfkL5vxD90s3Ej+ioeSK0BeqaO6Tf/lnkU2AaqpJolJC+C35/whQ4d051vhUqefPN3zF0MSXrf8rYgHt59rDIwzumbJQ3+x3g+8UOwWqdhIwrt3lVWdnYaDzkIGVIys6NoUpeDLl2RvWRKT7/8SQAJ2GKS1KffzMlnk5hAP5O2RzgeB5o9n1I/hZKoS44PIPZwRzmkbl0r8wE7XRhvwgOcKxNY1lqsTpVMR2CQrNBvqeU/mBpREjsWV33s+kctl9qd/DbHVpFRGMnGbRPe8zgUNF+spr7DU5p4343KMuuuE2V9m8VGM0vI3z9ZPgVqI6dRju3M2hFMxpX2k4iVQK+mv9rvx0WkvYUEvrFE3v0D1IQ3BgOmC+cykdxeTRZPZfptNXSVQWDfO9ihKUakUyICZzO/4HrxhfUqQGcFhNmwbVj2VQW/9vFhew8CCLzfAUIzLYb+iyEsO57CHf1/5a4u3ThANUUI9LCEHQIXzhLSqdF+H3ES4LwdzTNeHxcFMShEyPs0iD71tYUXX14W6/fpI1fJukbaa1C8Af/4OKNcXS6s/88PIVQ38Jda+tp6G4BQyQAR1raTZGUMmySuvM9M=
```
Comment (viren-nadkarni) [inline .travis.yml]: is this the token? is it okay to put it here?

File: Makefile [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
 
 SOURCES := speechmatics/ setup.py
+VERSION ?= $(shell cat VERSION).dev0
 
 all: lint test docs
 
 lint:
 	pylint $(SOURCES)
 	flake8 $(SOURCES)
+	pycodestyle $(SOURCES)
```
Comment (viren-nadkarni) [inline Makefile]: doesn't flake8 also wrap around pycodestyle?

File: README.md [hunk 3]
```diff
@@ -80,4 +84,4 @@ If you have any issues with this library or encounter any bugs then please get i
 
 ---
 
-License: MIT
+License: [MIT](https://opensource.org/licenses/MIT)
```
Comment (viren-nadkarni) [inline README.md]: Can we make the link point to LICENSE.txt instead?

File: .travis.yml (hunk 1 above)
Comment (weakcamel) [reply .travis.yml]: It's a token, but encrypted by Travis. And it only gets properly decrypted within the context of original repository, so forks won't get it.

So yes, it's reasonably safe.

https://docs.travis-ci.com/user/deployment/pypi/

File: Makefile (hunk 2 above)
Comment (weakcamel) [reply Makefile]: I thought it did, yet, all of the line length issues were caught by `pycodestyle` when I ran it separately, flake8 said "all clear".

File: README.md (hunk 3 above)
Comment (weakcamel) [reply README.md]: yeah, good idea

Comment (weakcamel) [reply README.md]: done

File: Makefile (hunk 2 above)
Comment (viren-nadkarni) [reply Makefile]: hmm, funny, i suppose it doesn't hurt running it separately

**Pull request merged**